DEHPMatlab/ViewModel/Row/MappedElementDefinitionRowViewModel.cs

```diff
@@ -0,0 +1,143 @@
+
+        /// <summary>
+        /// Backing field for <see cref="IsValid"/>
+        /// </summary>
+        private bool isValid;
+
+        /// <summary>
+        /// Backing field for <see cref="Value"/>
+        /// </summary>
+        private object value;
+
+        /// <summary>
+        /// Initializes a new <see cref="MappedElementDefinitionRowViewModel"/>
+        /// </summary>
+        public MappedElementDefinitionRowViewModel()
+        {
+            this.WhenAnyValue(x => x.SelectedParameter,
+                x => x.SelectedOption,
+                x => x.SelectedState,
+                x => x.SelectedMatlabVariable).Subscribe(_ => this.VerifyValidity());
+        }
+
+        /// <summary>
+        /// Gets or sets the selected <see cref="Parameter"/> of this view model
+        /// </summary>
+        public Parameter SelectedParameter
+        {
+            get => this.selectedParameter;
+            set => this.RaiseAndSetIfChanged(ref this.selectedParameter, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the selected <see cref="Option"/> of this view model
+        /// </summary>
+        public Option SelectedOption
+        {
+            get => this.selectedOption;
+            set => this.RaiseAndSetIfChanged(ref this.selectedOption, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the selected <see cref="ActualFiniteState"/> of this view model
+        /// </summary>
+        public ActualFiniteState SelectedState
+        {
+            get => this.selectedState;
+            set => this.RaiseAndSetIfChanged(ref this.selectedState, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the selected <see cref="MatlabWorkspaceRowViewModel"/> of this view model
+        /// </summary>
+        public MatlabWorkspaceRowViewModel SelectedMatlabVariable
+        {
+            get => this.selectedMatlabVariable;
+            set => this.RaiseAndSetIfChanged(ref this.selectedMatlabVariable, value);
+        }
+
+        /// <summary>
+        /// Asserts the validity for the mapping of this <see cref="MappedElementDefinitionRowViewModel"/>
+        /// </summary>
+        public bool IsValid
+        {
+            get => this.isValid;
+            set => this.RaiseAndSetIfChanged(ref this.isValid, value);
+        }
+
+        /// <summary>
+        /// The value of the parameter for the selected <see cref="Option"/> and the selected <see cref="ActualFiniteState"/>
+        /// </summary>
+        public object Value
+        {
+            get => this.value;
+            set => this.RaiseAndSetIfChanged(ref this.value, value);
+        }
+
+        /// <summary>
+        /// Verify the validity of this <see cref="MappedElementDefinitionRowViewModel"/>
+        /// </summary>
+        private void VerifyValidity()
+        {
+            this.Value = this.SelectedParameter?.QueryParameterBaseValueSet(this.SelectedOption, this.SelectedState).ActualValue[0];
```

Comment:
be aware that this could throw an exception depending on whether the SelectedParameter is option dependent or state dependent and SelectedOption is null or SelectedState is null.

Reply:
also this isn't correct if the Parameter is holding something else than a scalar value